### PR TITLE
Update SDK to support Ruby 2.3+

### PR DESCRIPTION
**Pull request by osowskit**
**Issue Type:** Improvement

**Changes Included:**

Ruby 2.2 and earlier have been deprecated. This PR will update Rubocop/Travis CI to support the minimum version of Ruby 2.3.

- Update target Ruby version in `.rubocop.yml`, `.gemspec`, and `.travis.yml` to 2.3
- Run `rubocop --auto-correct` to fix common linting/performance/style warnings
  - `.freeze` on strings is no longer required
- Disabled a handful of rubocop violations (potentially revisit these post merge)
- Updated Copyright information to 2019

**Check List:**

- [x] Includes adequate test [coverage](https://github.com/twitterdev/twitter-ruby-ads-sdk/tree/master/spec) for changes made.
- [ ] ~Includes new or updated [documentation](http://twitterdev.github.io/twitter-ruby-ads-sdk/reference/index.html)~.

### CODE REVIEW

**Comment by tushdante** [comment]
This is awesome! Appreciate the work @osowskit ! Two things:

1. I'm assuming you've run the examples in the repo, just to confirm that these changes don't break anything
2. Can you edit the [gemspec](https://github.com/twitterdev/twitter-ruby-ads-sdk/blob/0d980dd1c6242cd2b89cc69dec806265c640c14d/twitter-ads.gemspec#L11) file and add yourself as an owner? Feel free to remove Jacob and Brandon

**Comment by osowskit** [comment]
>  I'm assuming you've run the examples in the repo, just to confirm that these changes don't break anything

Excellent suggestion. I'll take a look at these now.

**Comment by smaeda-ks** [comment]
@osowskit Any updates on this? I'd love to get this merged for sure!

**Comment by osowskit** [comment]
>  you've run the examples in the repo, just to confirm that these changes don't break anything

I was part way through running the examples and was slowed down by not having an account with some permissions for specific features. I don't believe I'll have time in the next week or so to work on this but could add anyone to the forked repo to help 💪 this through

File: lib/twitter-ads/account.rb [hunk 1]
```diff
@@ -77,7 +77,7 @@ def all(client, opts = {})
     #
     # @return [String] The object instance detail.
     def inspect
-      str = String.new("#<#{self.class.name}:0x#{object_id}")
+      str = +"#<#{self.class.name}:0x#{object_id}"
```
Comment (smaeda-ks) [inline lib/twitter-ads/account.rb]: Just a note for myself: https://ruby-doc.org/core-2.4.0/String.html#method-i-2B-40
https://freelancing-gods.com/2017/07/27/friendly-frozen-string-literals.html

**Comment by smaeda-ks** [comment]
example scripts test for this PR:

- [x] active_entities.rb (from https://github.com/twitterdev/twitter-ruby-ads-sdk/pull/195)
- [x] analytics.rb
- [x] audience_intelligence.rb
- [x] batch_request.rb
- [x] cards_fetch.rb * need some fix
- [x] manual_request.rb
- [ ] metric_filtering.rb => I don't know what to do with this.
- [x] poll_cards.rb
- [x] promoted_tweet.rb * need some fix
- [x] quick_start.rb
- [x] tailored_audience.rb => deprecated content, should be updated later
- [x] tweet_preview.rb => can be removed
- [x] tweet_previews.rb (from https://github.com/twitterdev/twitter-ruby-ads-sdk/pull/194)

I believe this can be merged. I'll open another PR once this is merged to fix some minor bugs before we release a new version.

File: lib/twitter-ads/account.rb (hunk 1 above)
Comment (osowskit) [reply lib/twitter-ads/account.rb]: I believe this was part of the 'rubocop --auto-correct' changes. Good to see that this was expected

**Pull request merged**